app/graphql/types/query_type.rb

```diff
@@ -34,6 +34,14 @@ class QueryType < Types::BaseObject
       argument :id, ID, required: true
     end
 
+    field :project_permissions_for_project, [ProjectPermissionsType], null: true do
```

Comment:
After replying your `Permission.valid_project_action?` comment, I'm starting to think that maybe permissions should not be nested under project because the permission layer feels adjacent to projects, and isn't even linked to the Project model.  What do you think about that?

Reply:
Whether or not they are linked via a relationship in the model, the way we interact with project permissions is almost always in the context of projects or users. So, to me, it totally makes sense to nest project permissions under projects and/or users. Additionally, it will also reduce the number of queries the front end.

But, if you think about it by having a `subject_id` field and a `subject` field on Permissions we essentially have a polymorphic relationship. Even though we don't really treat it like that.

Reply:
Marking as resolved because of our separate conversation, and just pushed some updates to the PR.